harness/determined/pytorch/_pytorch_context.py

```diff
@@ -1037,8 +1040,22 @@ def train_batch(self, batch, epoch_idx, batch_idx):
         return self._tbd_writer
 
     def _maybe_reset_tbd_writer(self) -> None:
-        if self._tbd_writer is not None:
+        """
+        Reset the TensorBoard writer (close so the next write will reopen a new file) if more than
+        1 second has elapsed since the last reset. This is to ensure writes < 1 second apart do
+        not overwrite the same file name generated by TensorBoard from the epoch second.
```

Comment:
Recommend rephrasing this to be more literally what you do in the code.

Or make the code more literally what you do in the docstring.

Because what the code is doing now is _not_ checking that subsequent resets occurs at least a second later; it is checking that the next reset occurs within a different integer second.

I don't care which you do but docstrings that offer close-but-not-correct explanations are of very limited use.